### ISSUE TITLE
Improve splashscreen by hiding webview until fully loaded.

### ISSUE THREAD

**Issue by LasseRosenow** [home-assistant/android#3142]
**Is your feature request related to a problem? Please describe.**

I have been working on improving the splash screen experience of this app in the past: See #1938
The idea was to have the android launch-screen match the in-app launch screen.

But this does not work anymore since android 12 or 13? Has this new automatic splash screen, that has constraints on how large the icon could be etc. And the html icon is larger.

**Describe the solution you'd like**

I think having the html version in sync with the android version is to complicated and could always break in the future.
This is why I want to separate this and let the android app draw the whole splash screen, skipping the html splash screen in total.

The procedure should look like this:
1. Native android splash screen drawn by the os
2. in-app splash-screen that exactly matches the os splash-screen (we can set icon and color for os-splash screen, to make this more solid)
3. the in-app splash-screen is behind the webview, which is transparent
4. We wait for some event like `theme-update` from the webview maybe call it `app-initialized` and then show the webview.

This way the app startup would be massively clean and not consist of so many blinking etc. anymore :)

**Describe alternatives you've considered, if any**
I have considered making the icon in the html smaller for android devices, but I think this is not a good solution.

**Comment by LasseRosenow** [comment]
If you think this is a good idea, I will make a PR for this when I have spare time, probably in the next few weeks (at most month)

**Comment by LasseRosenow** [comment]
Maybe the `homeassistant_started` event could be the correct one to listen for :thinking:

**Comment by jpelgrom** [comment]
While it would be nice to improve this 'handoff' from app to frontend, relying on events makes it easier for this to break/prevent the webview from showing at all if the event isn't emitted or received for whatever reason. Examples: the frontend's "can't connect" message and #2684 already trigger the app to show a dialog it can't connect while it can (and the webview shows useful stuff).

**Comment by LasseRosenow** [comment]
What about showing the webview after for example 5 seconds, if we don't get the event?

This would probably make it work how it should be in 99% of the cases, but still show the webview, if something does not work as expected. The user in this case would just have a bit longer loading time, which should be fine since this is not the usual case, but rather some error case, that would not happen all the time.

**Comment by dshokouhi** [comment]
> I think having the html version in sync with the android version is too complicated and could always break in the future.
> This is why I want to separate this and let the android app draw the whole splash screen, skipping the html splash screen in total.

Personally speaking I would prefer to show HA loading and error messages rather than ones from the app, for example what about when HA fails to load and has that retry button how will a user see that? Would it make more sense to update the HTML loading page in HA frontend to better match app loading behavior? Probably need to take iOS into account too as I think they also need to use a splash screen.

> Maybe the `homeassistant_started` event could be the correct one to listen for 🤔

This is only fired once when HA starts like after an update restart.

If we are to go by any event then I think the `connected` event makes the most sense.

https://developers.home-assistant.io/docs/frontend/external-bus#connection-status-update-connection-status

**Comment by jpelgrom** [comment]
> What about showing the webview after for example 5 seconds, if we don't get the event?

Adding a timeout value like that would help limit the issues I was describing 👍

> Would it make more sense to update the HTML loading page in HA frontend to better match app loading behavior?

If the aim is to have them match exactly that would be quite difficult due to things like icon shapes.

> If we are to go by any event then I think the `connected` event makes the most sense.

Doesn't this event only fire after the frontend has loaded, meaning this change would 'slow down' loading the _app_?

**Comment by dshokouhi** [comment]
> > If we are to go by any event then I think the `connected` event makes the most sense.
> 
> Doesn't this event only fire after the frontend has loaded, meaning this change would 'slow down' loading the _app_?

yes it does but I thought thats what was meant by "until fully loaded"

**Comment by LasseRosenow** [comment]
> Doesn't this event only fire after the frontend has loaded, meaning this change would 'slow down' loading the app?

Not really slow down, more like show it only after the app splash screen is over, since the event gets fired from the WebView and not the server, it is a local event and not via network or something, so it would be instantaneous.

But it is important to fire the event just at the very moment the launch screen is complete and not to wait until "everything" is loaded

**Comment by TimoPtr** [comment]
We decided to fully hide the frontend and show it when it's fully loaded. It has been made possible because of the recent refactor of the WebViewActivity.
